### PR TITLE
Fix crash when placing tunnel bore

### PR DESCRIPTION
**Pull request by liach**
This fixed a potential crash. Might relate to #1046, but I am not very sure.

### CODE REVIEW

**Comment by liach** [comment]
There are some overhauls on hopper minecarts. See #1041

File: src/main/java/mods/railcraft/common/carts/EntityCartChest.java [hunk 1]
```diff
@@ -47,11 +50,6 @@ public boolean doInteract(EntityPlayer player, ItemStack stack, EnumHand hand) {
     }
 
     @Override
-    public EntityMinecart.Type getType() {
-        return EntityMinecart.Type.CHEST;
-    }
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/carts/EntityCartChest.java:52]: Why was this removed?

Comment (liach) [reply src/main/java/mods/railcraft/common/carts/EntityCartChest.java:52]: This is intended to return `null`. The vanilla enum `EntityMinecart.Type` should be treated as deprecated under Forge, since its functionalities are covered by Forge-provided new methods. The reason not to throw an exception is because of the patch addressed [here](https://github.com/MinecraftForge/MinecraftForge/pull/3592/files#diff-7bc83f1d40c6fc8d3356e0ce7057cf9c).

Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/carts/EntityCartChest.java:52]: This specific cart is intended to be for all intents and purposes seen as identical to the vanilla version of the chest cart. This won't change that will it?

**Comment by liach** [comment]
Will fix tomorrow.
